controllers/ocsinitialization/clusterclaims.go

```diff
@@ -0,0 +1,137 @@
+	err := client.Get(ctx, types.NamespacedName{Name: ClusterClaimCRDName}, &crd)
+	if err != nil {
+		if errors.IsNotFound(err) {
+			return false, nil
+		}
+		return false, err
+	}
+
+	return true, nil
+}
+
+func (obj *ocsClusterClaim) ensureCreated(r *OCSInitializationReconciler, instance *ocsv1.OCSInitialization) (reconcile.Result, error) {
+	ctx := context.TODO()
+	if crdExists, err := doesClusterClaimCrdExist(ctx, r.Client); !crdExists {
+		if err != nil {
+			r.Log.Error(err, "An error has occurred while fetching customresourcedefinition", "CustomResourceDefinition", ClusterClaimCRDName)
+			return reconcile.Result{}, err
+		}
+		return reconcile.Result{}, nil
+	}
+
+	creator := ClusterClaimCreator{
+		Logger:  r.Log,
+		Context: ctx,
+		Client:  r.Client,
+		OcsInit: instance,
+	}
+
+	operatorNamespace, err := creator.getOperatorNamespace()
+	if err != nil {
+		r.Log.Error(err, "failed to get operator's namespace. retrying again")
+		return reconcile.Result{}, err
+	}
+
+	OdfInfoNamespacedName := types.NamespacedName{
+		Namespace: operatorNamespace,
+		Name:      storagecluster.OdfInfoConfigMapName,
+	}.String()
+
+	kubeconfig, err := clientcmd.BuildConfigFromFlags("", "")
+	if err != nil {
+		creator.Logger.Error(err, "Failed to get kubeconfig for ClusterClaim client.")
+		return reconcile.Result{}, err
+	}
+
+	client, err := clusterclientv1alpha1.NewForConfig(kubeconfig)
+	if err != nil {
+		creator.Logger.Error(err, "Failed to create ClusterClaim client.")
+		return reconcile.Result{}, err
+	}
+
+	cc := &clusterv1alpha1.ClusterClaim{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: OdfInfoNamespacedNameClaimName,
+		},
+		Spec: clusterv1alpha1.ClusterClaimSpec{
+			Value: OdfInfoNamespacedName,
+		},
+	}
+
+	existingClaim, err := client.ClusterV1alpha1().ClusterClaims().Get(creator.Context, OdfInfoNamespacedNameClaimName, metav1.GetOptions{})
+	if err != nil {
+		if !errors.IsNotFound(err) {
+			creator.Logger.Error(err, "failed to get clusterclaim", "ClusterClaim", OdfInfoNamespacedNameClaimName)
+			return reconcile.Result{}, err
+		}
+		_, err = client.ClusterV1alpha1().ClusterClaims().Create(creator.Context, cc, metav1.CreateOptions{})
+		if err != nil {
+			creator.Logger.Error(err, "failed to create clusterclaim", "ClusterClaim", OdfInfoNamespacedNameClaimName)
+			return reconcile.Result{}, err
+		}
+		creator.Logger.Info("created clusterclaim", "ClusterClaim", cc.Name)
+	}
+
+	if equality.Semantic.DeepEqual(existingClaim.Spec, cc.Spec) {
+		creator.Logger.Info("clusterclaim unchanged", "ClusterClaim", OdfInfoNamespacedNameClaimName)
+	}
+
+	existingClaim.Spec = cc.Spec
+	_, err = client.ClusterV1alpha1().ClusterClaims().Update(creator.Context, existingClaim, metav1.UpdateOptions{})
+	if err != nil {
+		creator.Logger.Error(err, "failed to update clusterclaim", "ClusterClaim", OdfInfoNamespacedNameClaimName)
+		return reconcile.Result{}, err
+	}
+	creator.Logger.Info("updated clusterclaim", "ClusterClaim", cc.Name)
```

Comment:
This entire code can be replaced by a simple `CreateOrUpdate` call which will handle internally the differentiation between `Create`, `Update` and noop